ecommerce/core/management/commands/backfill_history.py

```diff
@@ -0,0 +1,193 @@
+            yield ids[i:i + chunk_size]
+
+    def replace_values(self, values, original, replacement):
+        values = [[replacement if v == original else v for v in value] for value in values]
+        return values
+
+    def handle(self, *args, **options):
+        batch_size = options['size']
+        sleep_between = options['sleep_between']
+        input_root = options['input_root']
+
+        for table_info in self.TABLES:
+            table = table_info['name']
+            historical_table = "_historical".join(table.rsplit('_', 1))
+            exclude_column = table_info['exclude_column']
+            input_filename = table_info['input_filename']
+            file_path = os.path.join(input_root, input_filename)
+            history_date = input_filename.rsplit('_')[-1]
+
+            with connection.cursor() as cursor:
+                query = u"""
+                    SELECT
+                        column_name
+                    FROM information_schema.columns
+                    WHERE table_name='{}'
+                    ORDER BY ordinal_position
+                    """.format(table)
+                cursor.execute(query)
+                columns = [column[0] for column in cursor.fetchall()]
+            if exclude_column in columns:
+                columns.remove(exclude_column)
+
+            # pylint: disable=open-builtin
+            with open(file_path, 'r') as input_file:
+                reader = csv.DictReader(input_file, delimiter='\x01')
+                lines = list(reader)
+
+                for rows in self.chunks(lines, batch_size):
+                    row_ids = [row['ID'] for row in rows]
+                    if table == 'courses_course':
+                        ids = ','.join("'{}'".format(id) for id in row_ids)
+                    else:
+                        ids = ','.join(row_ids)
+
+                    # Checks for existing historical records
+                    with connection.cursor() as cursor:
+                        query = u"""
+                            SELECT COUNT(1)
+                            FROM {historical_table}
+                            WHERE ID in ({ids})
+                            AND history_type='+'
+                            """.format(
+                                historical_table=historical_table,
+                                ids=ids
+                                )  # noqa
+                        cursor.execute(query)
+                        count = cursor.fetchone()[0]
+
+                    if count == len(rows):
+                        log.info(
+                            u"Initial history records already exist for ids %s..%s - skipping.",
+                            ','.join(row_ids[:2]), ','.join(row_ids[-2:])
+                        )
+                        continue
+                    elif count != 0:
+                        raise Exception(u"Database count: %s does not match input count: %s" % (count, len(ids)))
```

Comment:
ids is just a string, in think.  You probably want row_ids or rows:

```suggestion
                        raise Exception(u"Database count: %s does not match input count: %s" % (count, len(rows)))
```